frontend/src/App.js

```diff
@@ -1,102 +1,26 @@
-//     id: 2,
-//     name: ChuckNorris,
-//     alternate: 'Chuck Norris Quote',
-//     meal: 'BREAKFAST',
-//     title: "I am Chuck's title!",
-//     text: "I am Chuck's text!",
-//   },
-//   {
-//     id: 3,
-//     name: SnapeVoice,
-//     alternate: 'Snape v Potter',
-//     meal: 'BRUNCH',
-//     title: "I am Snape's title!",
-//     text: "I am the Snape's text!",
-//   },
-//   {
-//     id: 4,
-//     name: CatReligion,
-//     alternate: 'Cat Religion',
-//     meal: 'LUNCH',
-//     title: "I am the Cat Religion's title!",
-//     text: "I am the Cat Religion's text!",
-//   }
-// ]
 
 class App extends Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      message: 'Loading...'
-    }
-  }
-
-  componentDidMount() {
-    axios.get('/api/message.json')
-    .then(response => {
-      this.setState({ message: response.data.message })
-    })
-    .catch(console.error)
-  }
   render() {
     return (
-      <Router history={history}>
-          <div className="wrapper">
-            <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
-              <Link to="/"><h1 className="navbar-brand">{this.state.message}</h1></Link>
-              <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation">
-                <span className="navbar-toggler-icon"></span>
-              </button>
-              <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
-                <div className="navbar-nav">
-                  <Link to="/footer" className="nav-item nav-link active">Render Footer </Link>
-                  <Link to="/login" className="nav-item nav-link active">Render Login </Link>
-                  <Link to="/" className="nav-item nav-link active">Remove Components</Link>
-                </div>
-              </div>
-            </nav>
-
-          <Header/>
-          <MainSearch/>
-
-          <div className="wrapper">
-            <Route exact path="/" />
-            <Route path="/footer" component={MainFooter} />
-            <Route path="/login" component={LoginPage} />
-            <Route path="/signup" component={SignUp} />
-            <Route exact path="/users" history={history} render={(props) => <UsersAll {...props} />} />
-            <Route exact path="/users/:id" history={history} render={(props) => <User {...props} />} />
-          </div>
-          <MainFooter/>
+      <BrowserRouter>
+        <div>
+          <Switch>
+            <Route exact path='/' component={Homepage}></Route>
+            <Route path='/login' component={LoginPage}></Route>
+            <Route path='/signup' component={SignUp}></Route>
+            <Route exact path='/users' history={history} render={(props) => <UsersAll {...props} />}></Route>
+            <Route exact path='/users/:id' history={history} render={(props) => <User {...props} />}></Route>
```

Comment:
I do think it's OK to remove the `history` attribute here. Since these are `react-router-dom` routes, they should have access to `this.props.history`.